pytorch_lightning/accelerators/ddp_spawn_backend.py

```diff
@@ -64,12 +70,109 @@ def __recover_child_process_weights(self, model, best_path, last_path):
+        """
+        Entry point for ddp
+
+        Args:
+            process_idx:
+            mp_queue: multiprocessing queue
+            model:
+
+        Returns:
+
+        """
+        # show progressbar only on progress_rank 0
+        if (self.trainer.node_rank != 0 or process_idx != 0) and self.trainer.progress_bar_callback is not None:
+            self.trainer.progress_bar_callback.disable()
+
+        # determine which process we are and world size
+        self.set_world_ranks(process_idx)
+
+        # set warning rank
+        rank_zero_only.rank = self.trainer.global_rank
+
+        # set up server using proc 0's ip address
+        # try to init for 20 times at max in case ports are taken
+        # where to store ip_table
+        model.trainer = self.trainer
+        model.init_ddp_connection(
+            self.trainer.global_rank,
+            self.trainer.world_size,
+            self.trainer.is_slurm_managing_tasks
+        )
+
+        # call setup after the ddp process has connected
+        self.trainer.call_setup_hook(model)
+
+        # on world_size=0 let everyone know training is starting
+        if self.trainer.is_global_zero and not torch.distributed.is_initialized():
+            log.info('-' * 100)
+            log.info(f'distributed_backend={self.trainer.distributed_backend}')
+            log.info(f'All DDP processes registered. Starting ddp with {self.trainer.world_size} processes')
+            log.info('-' * 100)
+
+        # call sync_bn before .cuda(), configure_apex and configure_ddp
+        if self.trainer.sync_batchnorm:
+            model = model.configure_sync_batchnorm(model)
+
+        # move the model to the correct device
+        self.model_to_device(model, process_idx)
+
+        # CHOOSE OPTIMIZER
+        # allow for lr schedulers as well
+        self.setup_optimizers(model)
+
+        # set model properties before going into wrapper
+        self.trainer.model_connector.copy_trainer_model_properties(model)
+
+        # 16-bit
+        model = self.trainer.precision_connector.connect(model)
+
+        # device ids change depending on the DDP setup
+        device_ids = self.get_device_ids()
+
+        # allow user to configure ddp
+        model = model.configure_ddp(model, device_ids)
+
+        # set up training routine
+        self.trainer.train_loop.setup_training(model)
+
+        # train or test
+        results = self.train_or_test()
+
+        # get original model
+        model = self.trainer.get_model()
+
+        # persist info in ddp_spawn
+        self.transfer_distrib_spawn_state_on_fit_end(model, mp_queue, results)
+
+    def training_step(self, args):
+        if self.trainer.amp_backend == AMPType.NATIVE:
+            with torch.cuda.amp.autocast():
+                output = self.trainer.model(*args)
+        else:
+            output = self.trainer.model(*args)
+        return output
+
+    def validation_step(self, args):
+        output = self.training_step(args)
```

Comment:
Again, can we not call `training_step` here but use some other common function. Overwriting the `training_step` can yield unexpected behaviour here since I'd not expect `training_step` to influence `validation_step` and `test_step` at all!